server/controllers/userController.js

```diff
@@ -289,37 +289,40 @@ class UserController {
   }
 
   static async getProfile(req, res) {
-    const { isOwnProfile } = req;
-    const { id } = req.query;
+    const { isOwnProfile, loggedinUser } = req;
+    const { userId } = req.query;
 
     const returnUserObj = (userID) => {
       return models.Users.findByPk(userID, {
         attributes: ['profilePic', 'bio', 'favoriteQuote', 'favoriteBooks', 'firstName', 'lastName']
       })
     }
-    const user = await returnUserObj(id);
-      return util.successStatus(res, 200, 'profile retrieved successfully', {
-        isOwnProfile,
-        firstName: user.firstName,
-        lastName: user.lastName,
-        bio: user.bio,
-        profilePic: user.profilePic,
-        favoriteQuote: user.favoriteQuote,
-        favoriteBooks: user.favoriteBooks
-      })
+
+    let user;
+    if(userId){
+      user = await returnUserObj(userId);
+    }
+    else user = await returnUserObj(loggedinUser.id);
+    return util.successStatus(res, 200, 'profile retrieved successfully', {
```

Comment:
@encodedBicoding You didn't account for when the profile doesn't exist

Reply:
I'll get on it now.